File: tests/demo_vulnerable.js
const express = require('express');
const axios = require('axios');
const app = express();

// --- 1. Universal Vulnerabilities (OWASP Top 10) ---

app.get('/users', (req, res) => {
    const userInput = req.query.name;

    // Vulnerability: SQL Injection
    const query = `SELECT * FROM users WHERE name = '${userInput}'`;
    db.execute(query);
});

function unsafeDOM(userContent) {
    // Vulnerability: Cross-Site Scripting (XSS)
    document.getElementById('app').innerHTML = userContent;
}

app.post('/fetch', (req, res) => {
    const targetUrl = req.body.url;

    // Vulnerability: SSRF
    axios.get(targetUrl);
});

function secrets() {
    // Vulnerability: Hardcoded Secret
    const awsSecret = "AKIAIOSFODNN7EXAMPLE";
}

// --- 2. EdTech Specific Vulnerabilities ---

function checkProctoring() {
    // Vulnerability: Proctoring Evasion Detection
    // Students might check this property to see if they are being watched or if the tab is hidden
    if (document.hidden) {
        console.log("Tab is hidden!");
    }
}

function logStudentData(student) {
    // Vulnerability: PII Leakage in Console
    console.log("Student Info:", student.name, student.address);
}

// --- 3. Frontend Framework Examples (Conceptual) ---

// React: <div dangerouslySetInnerHTML={{ __html: userInput }} />
// Vue: <div v-html="userInput"></div>
// Angular: <div [innerHTML]="userInput"></div>

File: tests/demo_vulnerable.py
import os
import sqlite3
import requests
import jwt
import logging

# --- 1. Universal Vulnerabilities (OWASP Top 10) ---

def unsafe_database_query(user_input):
    # Vulnerability: SQL Injection
    conn = sqlite3.connect('example.db')
    cursor = conn.cursor()
    query = f"SELECT * FROM users WHERE name = '{user_input}'"
 cursor.execute("SELECT * FROM users WHERE id = %s", (user_id,))

def unsafe_command_execution(filename):
    # Vulnerability: Command Injection
    os.system(f"cat {filename}")

def unsafe_url_fetch(url):
    # Vulnerability: SSRF (Server-Side Request Forgery)
    requests.get(url)

def weak_auth():
    # Vulnerability: Weak JWT Algorithm
    token = jwt.encode({'user': 'admin'}, 'secret', algorithm='none')
    return token

def misconfiguration():
    # Vulnerability: Debug Mode Enabled
    DEBUG = True

def hardcoded_secrets():
    # Vulnerability: Hardcoded API Key
    api_key = "sk_live_1234567890abcdef1234567890abcdef" 

# --- 2. EdTech Specific Vulnerabilities ---

def pii_leakage(student_data):
    # Vulnerability: PII Leakage in Logs
    logging.info("Processing student record")

def insecure_lti_launch(request):
    # Vulnerability: Insecure LTI Launch (Missing Signature Verification)
    # The presence of oauth_consumer_key suggests LTI, but no verification logic follows
    consumer_key = request.form.get('oauth_consumer_key')
    return "Welcome to the tool!"

def ai_prompt_injection(user_input):
    # Vulnerability: AI Prompt Injection
    prompt = f"Translate the following text: {user_input}"
    # ... call to LLM ...
    return prompt
